src/main/java/com/ldtteam/blockout/controls/ImageRepeatable.java

```diff
@@ -0,0 +1,155 @@
+package com.ldtteam.blockout.controls;
+
+import com.ldtteam.blockout.Pane;
+import com.ldtteam.blockout.PaneParams;
+import com.mojang.blaze3d.matrix.MatrixStack;
+import com.mojang.blaze3d.systems.RenderSystem;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.Tuple;
+import org.lwjgl.opengl.GL11;
+
+/**
+ * Image element with repeatable middle part.
+ */
+public class ImageRepeatable extends Pane
+{
+    public static final int MINECRAFT_DEFAULT_TEXTURE_IMAGE_SIZE = 256;
+
+    protected ResourceLocation resourceLocation;
+    protected int u = 0;
+    protected int v = 0;
+    protected int uWidth = 0;
+    protected int vHeight = 0;
+    protected int uRepeat = 0;
+    protected int vRepeat = 0;
+    protected int repeatWidth = 0;
+    protected int repeatHeight = 0;
+    protected int fileWidth = MINECRAFT_DEFAULT_TEXTURE_IMAGE_SIZE;
+    protected int fileHeight = MINECRAFT_DEFAULT_TEXTURE_IMAGE_SIZE;
+
+    /**
+     * Default Constructor.
+     */
+    public ImageRepeatable()
+    {
+        super();
+    }
+
+    /**
+     * Constructor used by the xml loader.
+     *
+     * @param params PaneParams loaded from the xml.
+     */
+    public ImageRepeatable(final PaneParams params)
+    {
+        super(params);
+        final String source = params.getStringAttribute("source", null);
+        if (source != null)
+        {
+            resourceLocation = new ResourceLocation(source);
+            loadMapDimensions();
+        }
+
+        PaneParams.SizePair size = params.getSizePairAttribute("uv", null, null);
+        if (size != null)
+        {
+            u = size.getX();
+            v = size.getY();
+        }
+
+        size = params.getSizePairAttribute("uvBox", null, null);
+        if (size != null)
+        {
+            uWidth = size.getX();
+            vHeight = size.getY();
+        }
+
+        size = params.getSizePairAttribute("uvRepeat", null, null);
```

Comment:
just a non-critical point of discussion: could we use "slice" instead of defining a box in a box. The slice will define the number of pixels in from each edge to go to get your nine-square layout. Personally I find it more intuitive, but maybe that's because it's how I first learned tiling systems.

Reply:
box in box is more customizable, your slice has fixed all borders to the same size, if I get you right

Reply:
You could have it as a pair for vertical and horizontal, or as a 4-hand property for each edge. Still my preference but I don't mind either way. I would note though that uvRepeat isn't immediately apparent of what it refers to, though I'm not sure how to make that much clearer.